### PR TITLE
Further work on making azure actually do the right thing for staged recipes

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This intentionally fails on CI to show that things are actually working correctly for Azure.

Once we plan on merging I'll remove the bad recipe

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
This PR should fail CI presently

**Comment by mariusvniekerk** [comment]
Okay this is now doing the correct thing.  I'll rebase this and clean it up a bit.

File: .ci_support/compute_build_graph.py [hunk 1]
```diff
@@ -285,9 +286,17 @@ def add_intradependencies(graph):
         # what the build and host platforms are on the build machine.
         # However, all we know right now is what machine we're actually
         # on (the one calculating the graph).
+
+        test_requires = m.meta.get('test', {}).get('requires', [])
+
+        log.warn("node: {}".format(node))
```
Comment (isuruf) [inline .ci_support/compute_build_graph.py]: Why warn?

File: .ci_support/compute_build_graph.py [hunk 2]
```diff
@@ -285,9 +286,17 @@ def add_intradependencies(graph):
...
+        log.warn("   build: {}".format(m.ms_depends('build')))
+        log.warn("   host: {}".format(m.ms_depends('host')))
+        log.warn("   run: {}".format(m.ms_depends('run')))
+        log.warn("   test: {}".format(test_requires))
+
         deps = set(m.ms_depends('build') + m.ms_depends('host') + m.ms_depends('run') +
-                   [conda_interface.MatchSpec(dep) for dep in
-                    m.meta.get('test', {}).get('requires', [])])
+                   [conda_interface.MatchSpec(dep) for dep in test_requires or []])
```
Comment (isuruf) [inline .ci_support/compute_build_graph.py]: Why this change?

File: .ci_support/compute_build_graph.py (hunk 1 above)
Comment (mariusvniekerk) [reply .ci_support/compute_build_graph.py]: i can move this to info.  was useful debugging things

File: .azure-pipelines/azure-pipelines-osx.yml [hunk 3]
```diff
@@ -38,7 +35,7 @@ jobs:
   - script: |
       export PATH=$(Build.StagingDirectory)/miniconda/bin:$PATH
       set -x -e
-      conda install --yes --quiet conda-forge::conda-forge-ci-setup=2 networkx
+      conda install --yes --quiet conda-forge::conda-forge-ci-setup=2 networkx conda-forge-pinning
```
[the remark this answers is not in the document]
Comment (mariusvniekerk) [reply .azure-pipelines/azure-pipelines-osx.yml]: Fixed

File: .ci_support/build_all.py [hunk 4]
```diff
@@ -98,6 +98,7 @@ def build_folders(recipes_dir, folders, arch, channel_urls):
 
     worker = {'platform': platform, 'arch': arch,
               'label': '{}-{}'.format(platform, arch)}
+    print(worker)
```
Comment (isuruf) [inline .ci_support/build_all.py]: This is unnecessary. You can already see this information,
```
Computed that there are 3 distributions to build from 1 recipes
Resolved dependencies, will be built in the following order:
     psutil-5.5.0-python_2.7-on-linux-64
     psutil-5.5.0-python_3.6-on-linux-64
     psutil-5.5.0-python_3.7-on-linux-64
```

Comment (mariusvniekerk) [reply .ci_support/build_all.py]: fixed

**Comment by isuruf** [comment]
Why duplicate the file `.azure-pipelines/build_steps.sh` and `.circleci/build_steps.sh` ? Can one of them be a symlink to the other?

**Comment by mariusvniekerk** [comment]
they aren't exactly identical

**Comment by isuruf** [comment]
They can be identical. Sent https://github.com/mariusvniekerk/staged-recipes/pull/2

**Comment by mariusvniekerk** [comment]
Thanks!, now we can have staged-recipes azure reporting that isn't just lies

File: .circleci/build_steps.sh [hunk 5]
```diff
@@ -28,6 +28,7 @@ cp -r /home/conda/staged-recipes/.ci_support ~/.ci_support
 # Find the recipes from master in this PR and remove them.
 echo "Finding recipes merged in master and removing them from the build."
 pushd /home/conda/staged-recipes/recipes > /dev/null
+git fetch --force origin master:master
```
Comment (isuruf) [inline .circleci/build_steps.sh]: This seems to fail circleci
